ARCs/arc-0056.md

```diff
@@ -0,0 +1,381 @@
+     * included, but the corresponding genesis hash key MUST also be defined
+     */
+    [network: string]: {
+      /** The app ID of the deployed contract in this network */
+      appID: number;
+    };
+  };
+  /** Named structs use by the application */
+  structs: { [structName: StructName]: StructFields };
+  /** All of the methods that the contract implements */
+  methods: Method[];
+  state: {
+    /** Defines the values that should be used for GlobalNumUint, GlobalNumByteSlice, LocalNumUint, and LocalNumByteSlice when creating the application  */
+    schema: {
+      global: {
+        ints: number;
+        bytes: number;
+      };
+      local: {
+        ints: number;
+        bytes: number;
+      };
+    };
+    /** Mapping of human-readable names to StorageKey objects */
+    keys: {
+      global: { [name: string]: StorageKey };
+      local: { [name: string]: StorageKey };
+      box: { [name: string]: StorageKey };
+    };
+    /** Mapping of human-readable names to StorageMap objects */
+    maps: {
+      global: { [name: string]: StorageMap };
+      local: { [name: string]: StorageMap };
+      box: { [name: string]: StorageMap };
+    };
+  };
+  /** Supported bare actions for the contract. An action is a combination of call/create and an OnComplete */
+  bareActions: {
+    /** OnCompletes this method allows when appID === 0 */
+    create: ("NoOp" | "OptIn" | "DeleteApplication")[];
+    /** OnCompletes this method allows when appID !== 0 */
+    call: (
+      | "NoOp"
+      | "OptIn"
+      | "CloseOut"
+      | "ClearState"
+      | "UpdateApplication"
+      | "DeleteApplication"
+    )[];
+  };
+  /** Information about the TEAL programs */
+  sourceInfo?: {
+    /** Approval program information */
+    approval: SourceInfo[];
+    /** Clear program information */
+    clear: SourceInfo[];
+  };
+  /** The pre-compiled TEAL that may contain template variables. MUST be omitted if included as part of ARC23 */
+  source?: {
+    /** The approval program */
+    approval: string;
+    /** The clear program */
+    clear: string;
+  };
+  /** The compiled bytecode for the application. MUST be omitted if included as part of ARC23 */
+  byteCode?: {
+    /** The approval program */
+    approval: string;
+    /** The clear program */
+    clear: string;
+  };
+  /** Information used to get the given byteCode and/or PC values in sourceInfo. MUST be given if byteCode or PC values are present */
+  compilerInfo?: {
+    /** The name of the compiler */
+    compiler: "algod" | "puya";
+    /** Compiler version information */
+    compilerVersion: {
+      major: number;
+      minor: number;
+      patch: number;
+      commit?: string;
+    };
+  };
+  /** ARC-28 events that MAY be emitted by this contract */
+  events?: Array<Event>;
+  /** A mapping of template variable names as they appear in the teal (not including TMPL_ prefix) to their respecive types and values (if applicable) */
+  templateVariables?: {
```

Comment:
Would it make sense to also allow these templateVariables as part of the networks section (along with app id)?

As part of building puya source maps we realised we need to know the template variables used for a particular contract as this can impact the PC locations, and one of the ideas we had was using ARC-56 as a source of these values.

Reply:
@daniel-makerx how important is this in the end with the new way you were thinking about things? Still useful?

Reply:
Not required for source maps if template values are allowed in constant blocks. 

But may still be potentially useful in some use cases to know by network the values used?
Happy to leave it out tho, could always be added with a subsequent ARC

Reply:
Agreed, let's leave it out for now